### DISCUSSION TITLE
Installing (not Targeting) on Raspberry Pi 4?

### DISCUSSION THREAD

**Discussion by thestumbler** [modm-io/modm#1101] [General]
Just curious if anyone has done this. It seems possible, without giving it any deep thought. 
-Chris

**Comment by salkinium** [comment]
Should be possible. I'm running native arm-none-eabi-gcc and Python on ARM64 (macOS), so ARM64 cross-compile ought to work on Linux too.

**Reply by thestumbler** [reply]
That's good to hear. Although it's not quite the "Yes, user SouthPolePenguinPainter has reported it's up and running on an RPi4 no worries" 8-)

**Comment by thestumbler** [comment]
Still haven't gotten around to testing this. I have, however, failed to get two different Ubuntu machines to work with MODM. The failure is related to OpenOCD and I've been too busy to track that down. 

I have been reasonably successful installing it on several Windows machines, in native Windows and in WSL. But the WSL also has an issue with OpenOCD, that I haven't been able to track down yet, either. 

Based on my hurried testing, the native Python installs on Windows (the official packs from the Python organization) just didn't work for me. At issue was when you PIP installed a tool meant to be run from the command line, as opposed to a module you would include in your Python programs, the tool wasn't available on the path. I'm sure that could be sorted out, but my solution was more drastic in the interest of moving things along. 

We just installed PYENV-WIN and setup Python that way. I normally use PYENV anyway for managing versions and to help isolate installed packages for different projects alongside VENV. That wasn't at play here, but rather the Python installed via PYENV just played nicer with the command line. It also seems to peacefully coexist with those Python packs downloaded from Python.org YMMV. 

When the dust settles, I may have some edits / updates for the installation guide.

**Reply by salkinium** [reply]
> I may have some edits / updates for the installation guide.

Yes, please! It's difficult to keep these guides up-to-date without user input.…
